cogs/listeners/member_events_cog.py

```diff
@@ -0,0 +1,25 @@
+from discord.ext import commands
+import discord
+
+from interfaces.supabase import SupabaseInterface
+
+class MemberEventsListener(commands.Cog):
+    def __init__(self, bot) -> None:
+        self.bot = bot
+        super().__init__()
+
+    @commands.Cog.listener("on_member_join")
+    async def on_member_join(self, member: discord.Member):
+        SupabaseInterface().updateContributor(member)
```

Comment:
Interfaces are generally meant to be extended and not make instances of. Maybe you can refactor to making a helper class and define static methods on it.

Reply:
refactored into helpers/supabaseClient